RELEASE.md

```diff
@@ -0,0 +1,19 @@
+# Release
+
+To release new version of Kuma follow the steps:
+
+1. Create a CHANGELOG.md.
+2. Create PR to [kuma.io website repository](https://github.com/Kong/kuma-website) making sure that all new features
```

Comment:
From this description it looks like that you create a PR for https://kuma.io in order to document new features.

While, in fact, the goal is to add download links and installation guide for the new version.

Reply:
rephrased so it's clear that this is for links. Added new step for double-checking that features are documented